contracts/timealarms/src/alarms.rs

```diff
@@ -30,28 +41,42 @@ impl TimeAlarms {
             return Err(ContractError::InvalidAlarm(time));
         }
         contract::validate_addr(&deps.querier, &address)?;
-        Self::TIME_ALARMS.add(deps.storage, address, time)?;
+        self.time_alarms.add(deps.storage, address, time)?;
         Ok(Response::new())
     }
 
     pub fn try_notify(
+        &self,
         storage: &mut dyn Storage,
         ctime: Timestamp,
         max_count: AlarmsCount,
     ) -> Result<Response, ContractError> {
-        let dispatcher = OracleAlarmDispatcher::new();
-        let dispatcher = Self::TIME_ALARMS.notify(storage, dispatcher, ctime, max_count)?;
+        let dispatcher = self
+            .time_alarms
+            .alarms_selection(storage, ctime)
+            .take(max_count.try_into()?)
+            .enumerate()
+            .try_fold(
+                OracleAlarmDispatcher::new(),
+                |dispatcher, (id, alarm)| -> Result<OracleAlarmDispatcher, ContractError> {
+                    dispatcher.send_to(id.try_into()?, alarm?.0)
```

Comment:
I believe we should stop using generated Id-s. Any constant would suffice.

Reply:
@go2bodhi I do not see it resolved

Reply:
forgot to commit